Cesium.CodeGen/Ir/Expressions/SubscriptingExpression.cs

```diff
@@ -35,7 +35,12 @@ public IExpression Lower()
     public IValue Resolve(IDeclarationScope scope)
     {
         if (_expression is not IdentifierExpression identifier)
-            throw new WipException(230, "Subscription supported only for IdentifierConstantExpression");
+        {
+            if (_expression is not IValueExpression IValueExpresssion)
+                throw new CompilationException($"{_expression} is not IValueExpression");

```

Comment:
This message would be visible to user (developer who wrote program). It is impolite expect that developer know our compiler in such details. Probably would be better if we have message like this
```suggestion
                throw new CompilationException($"{_expression} is not a value expression");

```
Also would be good to have tests on this functionality.

Reply:
My message is probably off, but for other reasons. I cannot imagine better expression just now. Maybe @ForNeVeR come up with better idea, or after we have tests something pop into my mind.